signer/src/bitcoin/utxo.rs

```diff
@@ -109,6 +109,70 @@ pub trait GetFees {
     fn get_fees(&self) -> Result<Option<Fees>, Error>;
 }
 
+/// Filter out the deposit requests that do not meet the amount or fee requirements.
+pub struct DepositFilter<'a> {
+    requests: &'a Vec<DepositRequest>,
+    minimum_fee: u64,
+    per_deposit_minimum: u64,
+    per_deposit_cap: u64,
+    max_mintable_cap: u64,
+    amount_to_mint: u64,
+}
+
+impl<'a> DepositFilter<'a> {
+    /// Create a new [`DepositFilter`] instance.
+    pub fn new(
+        requests: &'a Vec<DepositRequest>,
+        minimum_fee: u64,
+        per_deposit_minimum: u64,
+        per_deposit_cap: u64,
+        max_mintable_cap: u64,
+    ) -> Self {
```

Comment:
nit: wouldn't it be simpler to have this function take the limits struct for some of these? 

It also might make sense to implement `validate_deposit` on the `SbtcLimits` struct itself, since it can be viewed as a separate kind of check. And I would imagine we'd put a `validate_withdrawal` function there too. But this is also fine.

Reply:
I tried to move it in `SbtcLimits` like you suggested but, when I looked a bit at the things to move around to support also the filtering of withdrawals, it felt too much validation logic needed to be moved in the signer_state file, and I didn't like it 🥲. Eventually, I renamed the struct `DepositFilter` -> `SbtcRequestsFilter`, thinking that we can implement the withdrawals filter in there once we'll know move onto that

Reply:
Oh okay, yeah I can see how that can happen, makes sense.